### ISSUE TITLE
Possible race condition in restart runs

### ISSUE THREAD

**Issue by singhbalwinder** [E3SM-Project/E3SM#2750]
One of my restart run (where `resubmit` in `env_run.xml` was > 0) crashed twice with the following error: 
```
doing an 6 nmonths startup test, with restarts every 6 nmonths
2019-02-15 00:42:10 CASE.RUN BEGINS HERE
2019-02-15 00:42:10 RUN_MODEL BEGINS HERE
2019-02-15 00:42:10 PRE_RUN_CHECK BEGINS HERE
2019-02-15 00:42:11 NAMELIST CREATION BEGINS HERE
Generating namelists for /lcrc/group/acme/bsingh/acme_scratch/anvil/
SMS_Ly5.ne30_ne30.FC5AV2C.anvil_intel.3rd_av2c__rebased_use_sgv_true

ERROR: File /lcrc/group/acme/bsingh/acme_scratch/anvil/
SMS_Ly5.ne30_ne30.FC5AV2C.anvil_intel.3rd_av2c__rebased_use_sgv_true/
env_run.xml appears to have changed without a corresponding invalidation, 
modtimes 1550212923.59 != 1550212930.82
```

It looks like a race condition where scripts modified env_run.xml before they should.

**Comment by jgfouca** [comment]
@singhbalwinder , can you give me a sequence of commands to reproduce this error?

**Comment by singhbalwinder** [comment]
I do not have a perfectly reproducible case as it happens randomly. It happened to me when I had a run which was resubmitting itself using env_run.xml's RESUBMIT functionality. The RESUBMIT functionality  changes env_run.xml file at each resubmit to change at least one field. It seemed like a race condition to me where a script  expects env_run.xml to stay the same while it is being modified by another script.

**Comment by jgfouca** [comment]
@singhbalwinder , I don't need perfect reproducibility, but I would still like the sequence of commands that can cause the error.

**Comment by jgfouca** [comment]
To clarify, the reason I'm pressing for details is because we do have tests in CIME for resubmission and I've never seen a race condition. Is it possible you had multiple CIME processes running simultaneously within a case?

**Comment by singhbalwinder** [comment]
I don't think so.  I think the first time I found out about this error was when I didn't find my job in queue. I checked the case directory to know what is going on and then I found this error. After that I got it, may be, 2-3 times and then I reported this here. I didn't get this error ever since. I think it might be a combination of machine acting weird which causes this race condition. 

May be we can close this issue for now and re-open it if it becomes more reproducible or bothersome in future....
